src/pokemon_icon.c

```diff
@@ -15,16 +16,1006 @@ struct MonIconSpriteTemplate
+    2, // Exploud
+    0, // Clamperl
+    0, // Huntail
+    0, // Gorebyss
+    0, // Absol
+    0, // Shuppet
+    0, // Banette
+    2, // Seviper
+    0, // Zangoose
+    1, // Relicanth
+    2, // Aron
+    2, // Lairon
+    2, // Aggron
+    0, // Castform
+    0, // Volbeat
+    2, // Illumise
+    2, // Lileep
+    0, // Cradily
+    0, // Anorith
+    0, // Armaldo
+    1, // Ralts
+    1, // Kirlia
+    1, // Gardevoir
+    2, // Bagon
+    2, // Shelgon
+    0, // Salamence
+    0, // Beldum
+    0, // Metang
+    0, // Metagross
+    2, // Regirock
+    2, // Regice
+    2, // Registeel
+    2, // Kyogre
+    0, // Groudon
+    1, // Rayquaza
+    0, // Latias
+    2, // Latios
+    0, // Jirachi
+    0, // Deoxys
+    0, // Chimecho
+    1, // Egg
+    0, // Unown B
+    0, // Unown C
+    0, // Unown D
+    0, // Unown E
+    0, // Unown F
+    0, // Unown G
+    0, // Unown H
+    0, // Unown I
+    0, // Unown J
+    0, // Unown K
+    0, // Unown L
+    0, // Unown M
+    0, // Unown N
+    0, // Unown O
+    0, // Unown P
+    0, // Unown Q
+    0, // Unown R
+    0, // Unown S
+    0, // Unown T
+    0, // Unown U
+    0, // Unown V
+    0, // Unown W
+    0, // Unown X
+    0, // Unown Y
+    0, // Unown Z
+    0, // Unown Exclamation Mark
+    0, // Unown Question Mark
+};
+
+static const struct SpritePalette sMonIconPaletteTable[] =
+{
+    { &gMonIconPalettes[0][0], POKE_ICON_BASE_PAL_TAG + 0 },
+    { &gMonIconPalettes[1][0], POKE_ICON_BASE_PAL_TAG + 1 },
+    { &gMonIconPalettes[2][0], POKE_ICON_BASE_PAL_TAG + 2 },
+
+// These palettes are actually part of the title screen BG palette and are not
+// used for Pokémon icons.
+    { &gMonIconPalettes[3][0], POKE_ICON_BASE_PAL_TAG + 3 },
+    { &gMonIconPalettes[4][0], POKE_ICON_BASE_PAL_TAG + 4 },
+    { &gMonIconPalettes[5][0], POKE_ICON_BASE_PAL_TAG + 5 },
+};
+
+static const struct OamData sMonIconOamData =
+{
+    .size = 2,
+    .priority = 1,
+};
+
+// fastest to slowest
+
+static const union AnimCmd sAnim_0[] =
+{
+    ANIMCMD_FRAME(0, 6),
+    ANIMCMD_FRAME(1, 6),
+    ANIMCMD_JUMP(0),
+};
+
+static const union AnimCmd sAnim_1[] =
+{
+    ANIMCMD_FRAME(0, 8),
+    ANIMCMD_FRAME(1, 8),
+    ANIMCMD_JUMP(0),
+};
+
+static const union AnimCmd sAnim_2[] =
+{
+    ANIMCMD_FRAME(0, 14),
+    ANIMCMD_FRAME(1, 14),
+    ANIMCMD_JUMP(0),
+};
+
+static const union AnimCmd sAnim_3[] =
+{
+    ANIMCMD_FRAME(0, 22),
+    ANIMCMD_FRAME(1, 22),
+    ANIMCMD_JUMP(0),
+};
+
+static const union AnimCmd sAnim_4[] =
+{
+    ANIMCMD_FRAME(0, 29),
+    ANIMCMD_FRAME(0, 29), // frame 0 is repeated
+    ANIMCMD_JUMP(0),
+};
+
+static const union AnimCmd *const sMonIconAnims[] =
+{
+    sAnim_0,
+    sAnim_1,
+    sAnim_2,
+    sAnim_3,
+    sAnim_4,
+};
+
+static const union AffineAnimCmd sAffineAnim_0[] =
+{
+    AFFINEANIMCMD_FRAME(0, 0, 0, 10),
+    AFFINEANIMCMD_END,
+};
+
+static const union AffineAnimCmd sAffineAnim_1[] =
+{
+    AFFINEANIMCMD_FRAME(0xFFFE, 0xFFFE, 0, 122),
```

Comment:
Well, these are actually fixed-point numbers. The value is really -2/256 or -0.0078125 in decimal. I'm not sure what the best way to represent these is.